src/backend/distributed/commands/index.c

```diff
@@ -522,6 +523,49 @@ GetCreateIndexRelationLockMode(IndexStmt *createIndexStatement)
 }
 
 
+/*
+ * ReindexStmtFindIndexOid returns the oid of the relation on which the index exist in reindex stmt.
+ * It also sets the relation given in the statement.
+ */
```

Comment:
Well, this is becoming a relatively dangerous function. We open a relation, and do not close it. 

Can we just return the relationOid and close the relation here as `PreprocessReindexStmt ` closes? And, the caller can re-open and close as it wishes?

Other alternative is that we return only the `relation` and the caller is responsible for closing; `OpenReindexStmtRelation(ReindexStmt, LockMode)`